Fix link format for installation guide in documentation

Hunk0724 opened the pull request:
## Description

**Why:**
The link to the "full guide" in `03a_contributors_quick_start_beginners.rst` was broken, preventing new contributors from accessing the necessary OS-specific Docker installation instructions.

**What:**
I corrected the link syntax in the document to `:doc:`full guide <03b_contributors_quick_start_seasoned_developers>`` to ensure the link correctly resolves to the `03b_contributors_quick_start_seasoned_developers.rst` file.

## Reviewer Note
The original broken link seemed to target `03b_contributors_quick_start_seasoned_developers.html`. Please confirm that linking to the source `.rst` file via the `:doc:` role is the correct approach, as this seems to be the intended destination for the full guide.

Hunk0724 committed:
Fix link format for installation guide in documentation

Lee-W commented on contributing-docs/03a_contributors_quick_start_beginners.rst:
```diff
@@ -56,7 +56,7 @@ For Breeze (local development):
 
 .. note::
-   Docker **or Podman** installation varies by OS. See the `full guide <03b_contributors_quick_start_seasoned_developers.html#local-machine-development>`_ for Ubuntu, macOS, and Windows instructions.
+   Docker **or Podman** installation varies by OS. See the `full guide <03b_contributors_quick_start_seasoned_developers.rst#local-machine-development>`_ for Ubuntu, macOS, and Windows instructions.
```
It was first introduced here: https://github.com/apache/airflow/pull/55930/files#diff-a1042dc8753272a9932e8599e84031efb7236b94e06c99770462dfc0dd16d87fR29. It's not totally wrong. However, GitHub cannot include `..include` correctly. Therefore, we should consider how to improve it further. Probably merge these?

Hunk0724 committed:
Update link to full guide to 03_contributors_quick_start.rst

Hunk0724 committed:
Update contributing-docs/README.rst

I would like to refactor for a better reading, based on the feedback regarding ..include issues, I've refactored the "New Contributor" content in the contributing-docs/README.rst

Hunk0724 committed:
Delete 03b_contributors_quick_start_seasoned_developers.rst

Because we have more detailed content in contributing-docs/README.rst
